restapi/api/reservation/reserve_endpoints.go

```diff
@@ -0,0 +1,138 @@
+	LastUsedTimestamp int64  `json:"lastUsed,omitempty"`
+}
+
+func CleanReservationsCRON() {
+	defer reserveMutex.Unlock()
+
+	// Every minute loop through the map of reserved devices and check if a reserved device last used timestamp was more than 5 minutes(300000 ms) ago
+	// If any, remove them from the map
+	for range time.Tick(time.Second * 60) {
+		reserveMutex.Lock()
+		for udid, reservedDevice := range reservedDevicesMap {
+			currentTimestamp := time.Now().UnixMilli()
+			diff := currentTimestamp - reservedDevice.LastUsedTimestamp
+
+			if diff > 300000 {
+				delete(reservedDevicesMap, udid)
+			}
+		}
+		reserveMutex.Unlock()
+	}
+}
+
+// Reserve device access
+// List          godoc
+// @Summary      Reserve a device
+// @Description  Reserve a device by provided UDID
+// @Tags         reserve
+// @Produce      json
+// @Success      200  {object} reservedDevice
+// @Router       /reserve/:udid [post]
+func ReserveDevice(c *gin.Context) {
+	udid := c.Param("udid")
+	reservationID := randomReservationID()
+
+	reserveMutex.Lock()
+	defer reserveMutex.Unlock()
+
+	// Check if there is a reserved device for the respective UDID
+	device := reservedDevicesMap[udid]
+	if device == nil {
+		newReservedDevice := reservedDevice{ReservationID: reservationID, LastUsedTimestamp: time.Now().UnixMilli()}
+		reservedDevicesMap[udid] = &newReservedDevice
+	} else {
+		c.IndentedJSON(http.StatusOK, reservedDevice{Message: "Already reserved"})
+		return
+	}
+
+	c.IndentedJSON(http.StatusOK, reservedDevice{ReservationID: reservationID})
+}
+
+// Release device access
+// List          godoc
+// @Summary      Release a device
+// @Description  Release a device by provided UDID
+// @Tags         reserve
+// @Produce      json
+// @Success      200  {object} reservedDevice
+// @Failure      404  {object} reservedDevice
+// @Router       /reserve/:udid [delete]
+func ReleaseDevice(c *gin.Context) {
+	udid := c.Param("udid")
+
+	reserveMutex.Lock()
+	defer reserveMutex.Unlock()
+
+	// Check if there is a reserved device for the respective UDID
+	device := reservedDevicesMap[udid]
+	if device == nil {
+		c.IndentedJSON(http.StatusNotFound, reservedDevice{Message: "Not reserved"})
+		return
+	} else {
+		delete(reservedDevicesMap, udid)
+		c.IndentedJSON(http.StatusOK, reservedDevice{Message: "Successfully released"})
+	}
+}
+
+// Get all reserved devices
+// List          godoc
+// @Summary      Get a list of reserved devices
+// @Description  Get a list of reserved devices with UDID, ReservationID and last used timestamp
+// @Tags         reserve
+// @Produce      json
+// @Success      200  {object} []reservedDevice
+// @Router       /reserved-devices [get]
+func GetReservedDevices(c *gin.Context) {
+	reserveMutex.Lock()
+	defer reserveMutex.Unlock()
+
+	var reserved_devices []reservedDevice
+	if len(reservedDevicesMap) == 0 {
+		c.IndentedJSON(http.StatusOK, reservedDevice{Message: "No reserved devices found"})
```

Comment:
I think it would be better to return `[]` empty list here for consistency

Reply:
done